test/weirollCall_test.js

```diff
@@ -0,0 +1,237 @@
+// @format
+import test from "ava";
+
+import {
+  concatio,
+  io,
+  command,
+  testLength,
+  flags,
+  CALLTYPES,
+  encodeFunctionCall,
+  call,
+  decodeParameters,
+} from "../src/index.js";
+
+test("produce actual weiroll static call for WETH contract", async (t) => {
+  const sel = Buffer.from("18160ddd", "hex"); // === keccak256("totalSupply()");
+  const f = Buffer.from("02", "hex");
+  const full = Buffer.from("ff", "hex");
+  const inp = Buffer.concat([full, full, full, full, full, full]);
+  const out = io(false, 0b0);
+  const target = Buffer.from("C02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2", "hex"); //WETH
+  const cmd = command(sel, f, inp, out, target);
+
+  const portalAddr = "0x6bb52BE538A1E5668a9c45626EDF2C590e86CA76";
+
+  const options = {
+    url: "https://cloudflare-eth.com",
+  };
+  const to = portalAddr;
+  // NOTE: We set msg.sender to address(0) as this is what the VM Portal
+  // contract expects: `if (msg.sender != caller) revert NotCaller();`
+  const from = "0x0000000000000000000000000000000000000000";
+  const data = encodeFunctionCall(
+    {
+      name: "execute",
+      type: "function",
+      inputs: [
+        {
+          type: "bytes32[]",
+          name: "commands",
+        },
+        {
+          type: "bytes[]",
+          name: "state",
+        },
+      ],
+    },
+    [[`0x${cmd.toString("hex")}`], []]
```

Comment:
The execution reversal stems from a problem with the `state` argument to weiroll's `CommandBuilder.writeOutputs`, at https://github.com/weiroll/weiroll/blob/main/contracts/CommandBuilder.sol#L138

It needs to have the same length as outputs, as far as I understand. Here, 1.

Here is a fix to this specific issue:

```suggestion
    [[`0x${cmd.toString("hex")}`], ['0x0']]
```

Reply:
let's fucking go, this solved it. Great work @lajarre!